packages/api/amplify_api/lib/src/method_channel_api.dart

```diff
@@ -284,55 +316,119 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
     }
   }
 
-  bool _shouldThrow(int statusCode) {
-    return statusCode < 200 || statusCode > 299;
-  }
-
-  RestResponse _formatRestResponse(Map<String, dynamic> res) {
+  AWSStreamedHttpResponse _formatRestResponse(Map<String, dynamic> res) {
     final statusCode = res['statusCode'] as int;
-    final headers = res['headers'] as Map?;
-    final response = RestResponse(
-      data: res['data'] as Uint8List?,
-      headers: headers?.cast<String, String>(),
-      statusCode: statusCode,
-    );
-    if (_shouldThrow(statusCode)) {
-      throw RestException(response);
-    }
-    return response;
+    // Make type-safe version of response headers.
+    final serializedHeaders = res['headers'] as Map?;
+    final headers = serializedHeaders?.cast<String, String>();
```

Comment:
callout: Do we want to do this some way that's safer? At one point I was creating a new map from this with only the string key/value pairs so it was type safe at runtime. This is obviously a little simpler.

Reply:
It's safe when used here since the native plugin only sends back string header values (it does the mapping in native code). Just not safe when creating request.